cdc: debezium protocol ddl events are missed by kafka consumer

### What did you do?

1. open  debezium protocol changefeed
2.  run workload

### What did you expect to see?

no error

### What did you see instead?

miss ddl events and dml events reported error
```
[2025/03/20 18:30:08.886 +08:00] [ERROR] [writer.go:178] ["error on running consumer"] [error="Failed query info: REPLACE INTO `workload`.`Data24` (`version`,`object_value`,`object_id`,`model_id`) VALUES (?,?,?,?); : [CDC:ErrMySQLTxnError]MySQL txn error: Error 1146 (42S02): Table 'workload.Data24' doesn't exist"] [errorVerbose="[CDC:ErrMySQLTxnError]MySQL txn error: Error 1146 (42S02): Table 'workload.Data24' doesn't exist\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20240318064555-6bd07397691f/errors.go:178\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20240318064555-6bd07397691f/normalize.go:175\ngithub.com/pingcap/tiflow/pkg/errors.WrapError\n\tgithub.com/pingcap/tiflow/pkg/errors/helper.go:34\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql.wrapMysqlTxnError\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql/mysql.go:814\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql.(*mysqlBackend).multiStmtExecute\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql/mysql.go:646\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql.(*mysqlBackend).execDMLWithMaxRetries.func1.4\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql/mysql.go:772\ngithub.com/pingcap/tiflow/cdc/sink/metrics.(*Statistics).RecordBatchExecution\n\tgithub.com/pingcap/tiflow/cdc/sink/metrics/statistics.go:82\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql.(*mysqlBackend).execDMLWithMaxRetries.func1\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql/mysql.go:740\ngithub.com/pingcap/tiflow/pkg/retry.run\n\tgithub.com/pingcap/tiflow/pkg/retry/retry_with_opt.go:57\ngithub.com/pingcap/tiflow/pkg/retry.Do\n\tgithub.com/pingcap/tiflow/pkg/retry/retry_with_opt.go:34\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql.(*mysqlBackend).execDMLWithMaxRetries\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql/mysql.go:721\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql.(*mysqlBackend).Flush\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/mysql/mysql.go:232\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn.(*worker).doFlush\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/worker.go:200\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn.(*worker).runLoop\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/worker.go:139\ngithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn.newSink.func1\n\tgithub.com/pingcap/tiflow/cdc/sink/dmlsink/txn/txn_dml_sink.go:118\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.10.0/errgroup/errgroup.go:78\nruntime.goexit\n\truntime/asm_amd64.s:1700\nFailed query info: REPLACE INTO `workload`.`Data24` (`version`,`object_value`,`object_id`,`model_id`) VALUES (?,?,?,?); "]
```

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```
master
v9.0.0
```

--- a/cmd/kafka-consumer/writer.go
+++ b/cmd/kafka-consumer/writer.go
@@ -474,7 +474,7 @@ func (w *writer) appendRow2Group(row *model.RowChangedEvent, progress *partition
 		return
 	}
 	switch w.option.protocol {
-	case config.ProtocolSimple, config.ProtocolOpen, config.ProtocolCanalJSON:
+	case config.ProtocolSimple, config.ProtocolOpen, config.ProtocolCanalJSON, config.ProtocolDebezium:
 		// simple protocol set the table id for all row message, it can be known which table the row message belongs to,
 		// also consider the table partition.
 		// open protocol set the partition table id if the table is partitioned.

--- a/pkg/sink/codec/debezium/codec.go
+++ b/pkg/sink/codec/debezium/codec.go
@@ -1135,6 +1135,7 @@ func (c *dbzCodec) EncodeDDLEvent(
 	switch e.Type {
 	case timodel.ActionCreateSchema,
 		timodel.ActionCreateTable,
+		timodel.ActionCreateTables,
 		timodel.ActionCreateView:
 		changeType = "CREATE"
 	case timodel.ActionAddColumn,

